extensions/amp-autocomplete/0.1/test/test-amp-autocomplete.js

```diff
@@ -25,66 +28,90 @@ describes.realWin(
     },
   },
   env => {
-    let win, doc, element, impl;
+    let win, doc, impl;
 
     beforeEach(() => {
       win = env.win;
       doc = win.document;
+      return buildAmpAutocomplete().then(ampAutocomplete => {
+        impl = ampAutocomplete;
+      });
+    });
 
+    function buildAmpAutocomplete(wantSsr) {
       const form = doc.createElement('form');
-      const ampAutocomplete = doc.createElement('amp-autocomplete');
-      ampAutocomplete.setAttribute('layout', 'container');
-      ampAutocomplete.setAttribute('filter', 'substring');
+      const element = createElementWithAttributes(doc, 'amp-autocomplete', {
+        layout: 'container',
+        filter: 'substring',
+      });
 
       const input = win.document.createElement('input');
       input.setAttribute('type', 'text');
-      ampAutocomplete.appendChild(input);
+      element.appendChild(input);
 
       const script = win.document.createElement('script');
       script.setAttribute('type', 'application/json');
       script.innerHTML = '{ "items" : ["apple", "banana", "orange"] }';
-      ampAutocomplete.appendChild(script);
+      element.appendChild(script);
 
-      form.appendChild(ampAutocomplete);
+      form.appendChild(element);
       doc.body.appendChild(form);
-      return ampAutocomplete.build().then(() => {
-        element = ampAutocomplete;
-        impl = element.implementation_;
+
+      if (wantSsr) {
+        element.removeAttribute('filter');
+        element.setAttribute('src', 'example.json');
+        const template = doc.createElement('template');
+        element.appendChild(template);
+      }
+
+      const ampAutocomplete = new AmpAutocomplete(element);
+      const ssrTemplateHelper = ampAutocomplete.getSsrTemplateHelper();
```

Comment:
Heh, now that we fixed `ssrTemplateHelper_` I noticed a lot of usage of other private vars in this test file. 

No action needed but note for the future.